### PR TITLE
Protects fields person resolver from access by non-current person

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

Currently, if you find a way to query for another user (for example, if you discover their Apollos ID) it's possible to get data on them that shouldn't be available (such as gender, email, and campus). 

This PR does two things to make this possible

1. Adds a helper function that can wrap resolver functions and stop them from returning data for users who aren't the current user. 

2. Caches the current user on the context to avoid duplicate requests.

### What design trade-offs/decisions were made?

n/a

### How do I test this PR?

1. Login as yourself in Graphiql, and find your own personalId
2. Run the following query, and ensure that you can see your user's details, but not Burke's details
```
{
  me: node(id:"MY PERSON ID HERE") {
    ... on  Person{
      id
      firstName
      lastName
      campus {
        name
      }
      gender
      email
    }
  }
  burke:node(id:"Person:9e584307fb4f32582830fea02470af67"){
    ... on  Person{
      id
      firstName
      lastName
      campus {
        name
      }
      gender
      email
    }
  }    
}
```

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by vinnyjth** [comment]
image

**Comment by vinnyjth** [comment]
This was requested by @calebpanza

**Comment by IsaacHardy** [comment]
@vinnyjth This looks good, but I think we should talk about exporting the UTILS file for use outside of the rock-connector. Currently, the UTILS methods are unusable when creating a new rock-connector from outside core. Exporting it similar to the way we export our data-sources could allow "churches" to pull in the UTILS methods in their own custom work.

File: packages/apollos-data-connector-rock/src/auth/__tests__/index.tests.js [hunk 1]
```diff
@@ -94,6 +94,26 @@ describe('Auth', () => {
...
+      });
+
+      context.dataSources.Auth.get = jest.fn(() => ({}));
+
+      context.currentPerson = { id: 123456 };
+
+      const result = await graphql(schema, query, rootValue, context);
+      expect(result).toMatchSnapshot();
+      expect(context.dataSources.Auth.get.mock).toMatchSnapshot();
+      expect(context.dataSources.Auth.get.mock.calls.length).toEqual(0);
```
Comment (IsaacHardy) [inline packages/apollos-data-connector-rock/src/auth/__tests__/index.tests.js]: Should we leave a comment here explaining what each expect test is "expecting"?

File: packages/apollos-data-connector-rock/src/people/resolver.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { createGlobalId } from '@apollosproject/server-core';
+import { enforceCurrentUser } from '../utils';
```
Comment (IsaacHardy) [inline packages/apollos-data-connector-rock/src/people/resolver.js]: Would be nice to be able to import this from the rock-connector itself.

Comment (vinnyjth) [reply packages/apollos-data-connector-rock/src/people/resolver.js]: 👍

File: packages/apollos-data-connector-rock/src/auth/__tests__/index.tests.js (hunk 1 above)
Comment (vinnyjth) [reply packages/apollos-data-connector-rock/src/auth/__tests__/index.tests.js]: 👍

**Comment by vinnyjth** [comment]
@IsaacHardy addressed both your changes

File: packages/apollos-data-connector-rock/src/index.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ import * as AuthSms from './auth-sms';
 import * as PersonalDevice from './personal-devices';
 import * as Template from './template';
 import * as Campus from './campuses';
+import * as Utils from './utils';
```
Comment (IsaacHardy) [inline packages/apollos-data-connector-rock/src/index.js]: https://media.giphy.com/media/D8vbqAZMb8WTm/giphy.gif

Comment (IsaacHardy) [reply packages/apollos-data-connector-rock/src/index.js]: Click it ^

**Comment by IsaacHardy** [comment]
After you fix the lint issue... :)

**Pull request merged**